rust/Makefile

```diff
@@ -126,7 +125,11 @@ quiet_cmd_rustc_procmacro = $(RUSTC_OR_CLIPPY_QUIET) P $@
 	mv $(objtree)/rust/$(patsubst lib%.so,%,$(notdir $@)).d $(depfile); \
 	sed -i '/^\#/d' $(depfile)
 
-$(objtree)/rust/libmodule.so: $(srctree)/rust/module.rs FORCE
+# Although `module` is a procedural macro and thus does not use `core.o`,
+# still list it as a dependency so `libmodule.so` gets automatically recompiled
+# when rustc version changes.
```

Comment:
Let's give the root reason:

```suggestion
# Procedural macros can only be used with the `rustc` that compiled it.
# Therefore, to get `libmodule.so` automatically recompiled when the compiler
# version changes, we add `core.o` as a dependency (even if it is not needed).
```

(I also wonder if we should detect Rust compiler changes in a different way, but it is OK for now.)